File: programmers/etc/스택&큐/기능개발.py
def solution(progresses, speeds):
    answer = []
    cnt = 0 # 완료된 기능
    days = 0 # 날짜

    while len(progresses) > 0 :
        if (progresses[0]+days*speeds[0]) >= 100 : # 완료
            progresses.pop(0) # 완료 -> 리스트에서 항목 제거
            speeds.pop(0)
            cnt += 1
        else :
            if cnt > 0 :
                answer.append(cnt)
                cnt = 0
            else :
                days += 1
    answer.append(cnt)
    return answer

print(solution([93, 30, 55], [1, 30, 5])) # [2,1]
print(solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1])) # [1, 3, 2]

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.io.*;
import java.util.*;

class Solution {
        public int[] solution(int[] progresses, int[] speeds) {
            int[] answer = {};
            // 결과를 담을 arrayList
            ArrayList<Integer> answerLst = new ArrayList<>();
            ArrayDeque<Integer> q = new ArrayDeque<>();

            // 남은 일수 계산 - 속도로 몇일나오는지 계산 (큐)
            for (int i=0; i<progresses.length; i++) {
                int day = 0;
                if ((100-progresses[i])%speeds[i] == 0) {
                    day = (100-progresses[i])/speeds[i];
                }
                else {
                    day = (100-progresses[i])/speeds[i] + 1;
                }
                q.offer(day);
            }

            int cnt = 1;
            int num = q.pollFirst(); // 일단 첫번째 값 기준 정하기 위해 q에서 poll -> 첫번째 남은 일 수

            // 비교
            while (true) {
                if (q.isEmpty()) break;
                if (num>=q.peekFirst()) { // 현재 작업의 남은 일수가 이전 작업의 남은 일수보다 작거나 같으면 같이 배포
                    q.pollFirst();
                    cnt +=1 ;
                } else if (q.peekFirst()>num){ // 현재 작업의 남은 일수가 이전 작업의 남은 일수보다 크면 배포 처리 => 새로운 기준 + cnt초기화
                    answerLst.add(cnt);
                    num = q.pollFirst();
                    cnt=1;
                }
            }
            answerLst.add(cnt); // 마지막으로 남은 작업 수

            // arrayList를 배열로 변환
            answer = new int[answerLst.size()];
            for (int i=0; i<answerLst.size(); i++) {
                answer[i] = answerLst.get(i);
            }
            return answer;
        }
    }
